dwave/optimization/src/graph.cpp

```diff
@@ -386,6 +393,56 @@ void Node::initialize_state(State& state) const {
     state[topological_index_] = std::make_unique<NodeStateData>();
 }
 
+// We try to avoid needing to override this method in every class by doing
+// some compiler-specific stuff.
+std::string Node::classname() const {
+    // get the compiler-specific name
+    const char* compiler_name = typeid(*this).name();
+
+#if defined(_HAS_CXXABI)
+    // if cxxabi.h is available, we can demangle
+    std::string name;
+
+    // __cxa_demangle requires us to manage the lifespan of the returned pointer
+    // so we do this a bit carefully
+    {
+        int status = 0;
+        std::size_t size = 0;
+        char* unmangled = abi::__cxa_demangle(compiler_name, NULL, &size, &status);
```

Comment:
I think you should ensure `status` is equal to 0 after this, otherwise you may be initializing `std::string` from a null pointer and/or trying to free a null pointer.